src/components/dialogs/create-case-dialog.tsx

```diff
@@ -132,78 +165,85 @@ const CreateCaseDialog: React.FC<CreateCaseDialogProps> = ({
+            elementType: ElementType.CASE,
+            setError,
+            clearErrors,
+        }
+    );
 
     useEffect(() => {
         setIsCreationAllowed(
-            !!caseFile && !caseNameError && !caseNameChecking && !caseFileError
+            !!caseFile &&
+                !caseNameErrorMessage &&
+                !caseNameChecking &&
+                !caseFileErrorMessage
         );
-    }, [caseFile, caseFileError, caseNameChecking, caseNameError]);
+    }, [
+        caseFile,
+        caseNameErrorMessage,
+        caseNameChecking,
+        caseFileErrorMessage,
+    ]);
 
     return (
-        <Dialog
-            fullWidth={true}
-            open={open}
-            onClose={handleCloseDialog}
-            aria-labelledby="create-case-form-dialog-title"
-        >
-            <DialogTitle id="create-case-form-dialog-title">
-                <FormattedMessage id="ImportNewCase" />
-            </DialogTitle>
-            <DialogContent>
-                <TextFieldInput
-                    label={'nameProperty'}
-                    value={caseName}
-                    setValue={setCaseName}
-                    error={!!caseFileError}
-                    autoFocus
-                    adornment={caseFileAdornment}
-                    setValueHasChanged={setCaseNameChanged}
-                />
-                <TextFieldInput
-                    label={'descriptionProperty'}
-                    value={description}
-                    setValue={setDescription}
-                />
-                {caseNameError && (
-                    <Alert severity="error">{caseNameError}</Alert>
-                )}
-                {caseFileError && (
-                    <Alert severity="error">{caseFileError}</Alert>
-                )}
-                <UploadNewCase
-                    caseFile={caseFile}
-                    handleCaseFileUpload={handleCaseFileUpload}
-                />
-            </DialogContent>
-            <DialogActions>
-                <Button onClick={handleCloseDialog} variant="text">
-                    <FormattedMessage id="cancel" />
-                </Button>
-                <Button
-                    onClick={handleCreateNewCase}
-                    disabled={!isCreationAllowed}
-                    variant="outlined"
-                >
-                    <FormattedMessage id="validate" />
-                </Button>
-            </DialogActions>
-        </Dialog>
+        <FormProvider {...createCaseFormMethods}>
+            <Dialog
+                fullWidth={true}
+                open={open}
+                onClose={handleCloseDialog}
+                aria-labelledby="create-case-form-dialog-title"
+            >
+                <DialogTitle id="create-case-form-dialog-title">
+                    <FormattedMessage id="ImportNewCase" />
+                </DialogTitle>
+                <DialogContent>
+                    <TextFieldInput
```

Comment:
Use one of the component we made that manage the form by itself
TextInput for instance

Reply:
done, thank you